hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/TrimmingStringConverter.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2023 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.utils;
+
+import org.springframework.util.StringUtils;
+
+import com.vaadin.data.Converter;
+import com.vaadin.data.Result;
+import com.vaadin.data.ValueContext;
+
+/**
+ * Converter to trim whitespaces for an input e.g. to further convert it into an
+ * Integer, etc.
+ */
+public class TrimmingStringConverter implements Converter<String, String> {
+    @Override
+    public Result<String> convertToModel(final String s, final ValueContext valueContext) {
+        return Result.ok(StringUtils.trimWhitespace(s));
+    }
+
+    @Override
+    public String convertToPresentation(final String s, final ValueContext valueContext) {
+        return StringUtils.trimWhitespace(s);
```

Comment:
Please check if we need this trim call here. I would assume that the value from the database is already trimmed and we don't need to do it again.

Reply:
Verified: Yes, it is trimmed. But we have a problem with null values here. In case the value is not present in the database, we should provide an empty string here which can be set as value on the TextField. Therefore I have extended the logic perform a null check as well.